tests/test_register_proxy.py

```
@@ -0,0 +1,109 @@
+    httpx_mock.add_response(
+        url=REGISTER_PROXY_URL,
+        status_code=401,
+        json={"error": "unauthorized"},
+    )
+
+    with pytest.raises(httpx.HTTPStatusError) as exc_info:
+        await register_to_proxy(
+            proxy_url=MOCK_PROXY_URL,
+            subject_index=MOCK_SUBJECT_INDEX,
+            public_url=MOCK_ENGINE_URL,
+            secret_key="wrong-secret",
```

Ruff (0.15.18)

[error] Possible hardcoded password assigned to argument: "secret_key"

(S106)